Sources/JWT/Signers/RSASigner+X509.swift

```diff
@@ -0,0 +1,30 @@
+import CTLS
+
+typealias CX509Key = UnsafeMutablePointer<X509>
+
+extension RSAKey {
+    public init(x509Cert cert: Bytes) throws {
+        guard
+            let cert = cert.withUnsafeBufferPointer({ rawKeyPointer -> CX509Key? in
+                var base = rawKeyPointer.baseAddress
+                let count = cert.count
+
+                return d2i_X509(nil, &base, count)
+            }),
+            let pubkey = X509_get_pubkey(cert),
+            let rsa = EVP_PKEY_get1_RSA(pubkey) else {
+                throw JWTError.createPublicKey
+        }
+
+        // free resources
+        EVP_PKEY_free(pubkey)
```

Comment:
I suggest putting this (and any other free operations) into `defer` blocks, so low-level memory will be cleaned up properly even in case of errors.